### DISCUSSION TITLE
Why are some TestStoreTests test cases wrapped in #if DEBUG?

### DISCUSSION THREAD

**Discussion by gilgtm** [pointfreeco/swift-composable-architecture#2015] [Q&A] [answered]
Hi, I'm scratching my head at [this conditional](https://github.com/pointfreeco/swift-composable-architecture/blob/main/Tests/ComposableArchitectureTests/TestStoreTests.swift#L87) that wraps some test cases in `if #DEBUG`. There's probably some simple explanation, but I'm hoping to learn 🙏

**Comment by mbrandonw** [comment, marked as answer]
Hi @gilgtm, for that test in particular it is because we are specifically testing that the `TestStore` causes a failure when a full state change is not asserted on. You can see that because we are using `XCTExpectFailure`.

However, due to a confluence of reasons, it is not possible for TCA to directly use `XCTFail` in library code, which is what led us to [open source a library](https://github.com/pointfreeco/xctest-dynamic-overlay) that dynamically loads it. And because it is dynamically loaded code, it is not App Store safe, hence we have to hide it behind `#if DEBUG`. This means when the test runs in release mode, the `TestStore` assertion does not happen, which means the `XCTExpectFailure` fails. So, we have to exclude those tests from release.
